cookbooks/fb_users/recipes/default.rb

```diff
@@ -19,6 +19,12 @@
 # limitations under the License.
 #
 
+if defined?(ChefSpec)
+  file 'test resource' do
```

Comment:
can we just define this in the spec vs directly in the recipe?

Reply:
I can find no way to do that - if yo know of one, I'm all ears. The `chef_run do ... end` doesn't have the recipe DSL in it, not does the `converge do ... end`. Most examples I find on the internet either do what I"m doing here, or they make a test recipes, that the include_recipe's the real recipe and they run the specs on the test recipe, but that seems worse to me.

Reply:
I was thinking about something like this: https://github.com/chefspec/chefspec/blob/master/examples/recipe_block/spec/recipe_block_spec.rb#L16-L29

Reply:
I see how you got there, but I don't think that works quite like you think it does,

We already _have_ a `recipe` block in the test, at the very top:
https://github.com/facebook/chef-cookbooks/blob/master/cookbooks/fb_users/spec/resource_spec.rb#L24

The `recipe` block can either be an inline recipe, like the one you link to, or it can be a recipe from your cookbook you want to test, as it is in this case:

```
recipe 'fb_users::default' do |tc|
```

But more importantly, we actually override the `recipe` keyword in the fbspec stubs so that it sets up all the tons and tons of boilerplate you have to setup to get all your platforms and recipes setup in chefspec which is really cumbersome, so `recipe` for us means something a bit different (roughly: loop over all our platforms - module `include` and `exclude` - and then setup the proper recipe block for each fauxhai profile).

Reply:
You can stitch in a 'real' recipe (e.g. fb_users::spec) at the `converge` statement, so have something like: `converge('fb_users::spec', described_recipe)`.  Having the separate "unit test" recipe feels lot cleaner to me versus polluting the primary implementation recipe.

Reply:
Oh neat, thanks @joshuamiller01 ! I'll give that a shot.